### ISSUE TITLE
`prefer-flat-map` fixer bug

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#315]
`Array#flat` accepts argument `depth`, but flatMap not.

The fixer is not considering this `depth`, codes like

```js
const foo = [[1]].map(i => [i]).flat(2)
```

should not be reported.

**Comment by fisker** [comment]
@MrHen
